### ISSUE TITLE
Using future margin

### ISSUE THREAD

**Issue by mapapuche** [ccxt/ccxt#8375]
- OS: Windows 10
- Programming Language version: Python
- CCXT version: 1.40.11

Hello, i'm trying to use ccxt to trade on binance future. I have lot of trouble with the doc to understand how it works :/

So, first i create_market_buy_order, that work well. Then i put my stoploss / takeprofit.
The problem is i don't know if i don't use correctly the library, or if i don't use correctly the parameters to create a TAKE_PROFIT.

 I have this error:

`Traceback (most recent call last):
  File "main.py", line 93, in <module>
    buy(binance, crypto, price, r1, s1)
  File "main.py", line 41, in buy
    print(exchange.create_order('{}/USDT'.format(crypto), 'STOP', 'sell', size_to_buy, params))
  File "/mnt/c/Projects/arbitrage/venv/lib/python3.8/site-packages/ccxt/binance.py", line 1655, in create_order
    request['price'] = self.price_to_precision(symbol, price)
  File "/mnt/c/Projects/arbitrage/venv/lib/python3.8/site-packages/ccxt/base/exchange.py", line 1285, in price_to_precision
    return self.decimal_to_precision(price, ROUND, self.markets[symbol]['precision']['price'], self.precisionMode, self.paddingMode)
  File "/mnt/c/Projects/arbitrage/venv/lib/python3.8/site-packages/ccxt/base/decimal_to_precision.py", line 55, in decimal_to_precision
    dec = decimal.Decimal(str(n))
decimal.InvalidOperation: [<class 'decimal.ConversionSyntax'>]`

Here my code

```
    exchange_class = getattr(ccxt, 'binance')
    binance = exchange_class({
        'apiKey': 'KEY',
        'secret': 'SECRET',
        'timeout': 3000,
        'enableRateLimit': True,
        'options': {
            'defaultType': 'future',  # ←-------------- quotes and 'future'
        },
    })
    
        crypto = 'KSM'
        size_to_buy = 2.8921478159368905
        params = {
        'quantity': size_to_buy,
        'price': float("{:0.0{}f}".format(r1 - (r1 * 10 / 100), 8)),
        'stopPrice': float("{:0.0{}f}".format(r1 - (r1 * 10 / 100), 8)),
        'type': 'TAKE_PROFIT',
    }
    binance.create_order('{}/USDT'.format(crypto), 'TAKE_PROFIT', 'sell', size_to_buy, params)
```

Can you help me please :s ?

**Comment by kroitor** [comment]
The order of arguments to `create_order` is:

-  `symbol, type, side, amount, price, params`

You are passing:

- `symbol, type, side, amount, params` ← your `params` go instead of `price

Make sure to follow the Manual and be precise with method arguments.

```Python
exchange_class = getattr(ccxt, 'binance')
exchange = exchange_class({
    'apiKey': 'KEY',
    'secret': 'SECRET',
    'enableRateLimit': True,
    'options': {
        'defaultType': 'future',
    },
})
exchange.load_markets()
symbol = 'CSM/USDT'
amount = 2.8921478159368905
price = None
stopPrice = YOUR_STOP_PRICE # 1.2345 for example
params = {
    'stopPrice': exchange.price_to_precision(symbol, stopPrice),
}
exchange.create_order(symbol, 'TAKE_PROFIT', 'sell', amount, price, params)
```

Let us know if that does not answer your question.

**Comment by mapapuche** [comment]
Oh i am stupid ... thank for the fast reply

**Comment by kroitor** [comment]
@mapapuche if you update to the most recent version of the lib, this will also work:

```Python
exchange_class = getattr(ccxt, 'binance')
exchange = exchange_class({
    'apiKey': 'KEY',
    'secret': 'SECRET',
    'enableRateLimit': True,
    'options': {
        'defaultType': 'future',
    },
})
exchange.load_markets()
symbol = 'CSM/USDT'
amount = 2.8921478159368905
price = YOUR_LIMIT_PRICE  # 543.21 for example
params = {
    'stopPrice': YOUR_STOP_PRICE, # 1.2345 for example
}
exchange.create_order(symbol, 'TAKE_PROFIT', 'sell', amount, price, params)
```

**Comment by mapapuche** [comment]
Ok thank you. I'll do it.

**Comment by mapapuche** [comment]
Hi again @kroitor 

sorry to distrub you again. I have some strange thing.

I'm retreiving data with ohlcv :
`        ohlcv = binance.fetch_ohlcv('REN/USDT', timeframe='5m', limit=10)
        print(ohlcv)`

My output is : 
`[[1611972900000, 0.57257, 0.57322, 0.56864, 0.57048, 66342.0], [1611973200000, 0.57079, 0.57482, 0.56984, 0.57444, 54880.0], [1611973500000, 0.57444, 0.57621, 0.57232, 0.57531, 49217.0], [1611973800000, 0.5753, 0.57714, 0.57422, 0.57422, 43809.0], [1611974100000, 0.57433, 0.57671, 0.57345, 0.57562, 25451.0], [1611974400000, 0.57548, 0.57627, 0.57361, 0.57381, 17515.0], [1611974700000, 0.57385, 0.57452, 0.57081, 0.57199, 53374.0], [1611975000000, 0.57203, 0.57294, 0.56901, 0.56935, 23069.0], [1611975300000, 0.56948, 0.57074, 0.56654, 0.56654, 47614.0], [1611975600000, 0.56682, 0.56899, 0.56426, 0.56427, 89998.0]]`


The close / open price are not coresponding at all with KSM right now.
Right now KSM is about 0.61XXX and not 0.5XXXX

Look like i retrtieve wrong period or maybe wrong crypto currencie. The thing very strange, is when i change timeframe that work sometime, and sometime not.

Any idea please  ?

**Comment by kroitor** [comment]
@mapapuche you are fetching REN/USDT and comparing it to KSM/USDT?

**Comment by mapapuche** [comment]
Sorry i misspell, i compare to REN/USDT

**Comment by mapapuche** [comment]
![image](https://user-images.githubusercontent.com/64193148/106396284-43cd2a80-6407-11eb-8f9a-b19a3358e6db.png)

![image](https://user-images.githubusercontent.com/64193148/106396313-63645300-6407-11eb-9e27-47911c5a7ab2.png)

**Comment by kroitor** [comment]
@mapapuche that's actually a bug in CCXT, it fetches 500 most recent candles and returns the first 10 of them, not the last 10. I've just pushed a fix for this issue, and it will arrive in ~20 minutes in version 1.41.32. Your code should work properly if you upgrade your version of CCXT when it arrives. Thank you for reporting it!

**Comment by mapapuche** [comment]
Ok ! Thank you a lot for your hard work and fast answer as always <3

**Comment by 304s** [comment]
Yup []()

**Comment by kroitor** [comment]
@mapapuche version 1.41.32 is available now, thx again for your feedback!

**Comment by yanis7774** [comment]
> 
> 
> @mapapuche that's actually a bug in CCXT, it fetches 500 most recent candles and returns the first 10 of them, not the last 10. 



> that's actually a bug in CCXT, it fetches 500 most recent candles

Hi! Is it possible to fetch more than 500 candles? Before I could do it, but now it is not working. Still working on spot but not on the futures.

**Comment by kroitor** [comment]
@yanis7774 hi! No, afaik Binance has limited the max number of candles to 500 on spot/futures markets.

**Comment by kroitor** [comment]
@yanis7774 you can still fetch the candles beyond 500 by using pagination as described in the Manual:

- https://github.com/ccxt/ccxt/wiki/
- https://github.com/ccxt/ccxt/wiki/Manual
- https://github.com/ccxt/ccxt/wiki/Manual#pagination

**Comment by yanis7774** [comment]
> 
> 
> @yanis7774 hi! No, afaik Binance has limited the max number of candles to 500 on spot/futures markets.

I still can get more than 500 candles in early version of the lib (1.39.41 e.g.)

**Comment by kroitor** [comment]
@yanis7774 yep, looks like Binance has fixed the default limit to 500 and the max limit to 1500, as documented in their docs. I've pushed a fix for that, so if you upgrade to 1.41.52+ you will be able to get up to 1500 candles. Thank you for pointing it out!
